frontends/p4/tableKeyNames.cpp

```diff
@@ -32,10 +38,16 @@ class KeyNameGenerator : public Inspector {
 
     void postorder(const IR::Expression* expression) override { error(expression); }
 
+    /** Compute a name annotation for @expression.  Eg. `@name("foo.bar")` for
```

Comment:
These comments on the postorder methods are practically useless.
This one in particular is also wrong, since a PathExpression cannot be "foo.bar".
The code of these methods is so short that it is self-explanatory; I don't think that these comments help at all.
What should be documented is the "name" field of the KeyNameGenerator; for each expression is maintains a cstring representing the tentative name of the expression. The name of an expression is built recursively from its components. if an expression does not have a name in this map, then it is too complex.